google-cloud-spanner/test/google/cloud/spanner/snapshot/range_test.rb

```diff
@@ -22,7 +22,7 @@
   let(:session) { Google::Cloud::Spanner::Session.from_grpc session_grpc, spanner.service }
   let(:transaction_id) { "tx789" }
   let(:transaction_grpc) { Google::Cloud::Spanner::V1::Transaction.new id: transaction_id }
-  let(:snapshot) { Google::Cloud::Spanner::Snapshot.from_grpc transaction_grpc, session }
+  let(:snapshot) { Google::Cloud::Spanner::Snapshot.from_grpc transaction_grpc, session, nil }
```

Comment:
nit: Why are we importing `nil` at every place?

Reply:
Because snapshot object now takes directed read options as a param from client which is not needed in the existing tests. Because of the existing structure couldn't add it as default value that will require a refactoring.